### ISSUE TITLE
[Feature Request]: Argmax and argmin functions for tuples and arrays

### ISSUE THREAD

**Issue by Iainmon** [chapel-lang/chapel#25659]
### Summary of Feature

**Description:**
Add argmax and argmin functions for arrays and homogeneous tuples. 

**Is this issue currently blocking your progress?**
No. I have come up with a solution, but it is linear time in the length of array and only works for rank = 1 arrays. No solution for tuple.

**Comment by bradcray** [comment]
Are argmax/argmin like Chapel's maxloc/minloc reductions?

**Comment by Iainmon** [comment]
Yes, it appears they are. Before I close, is the status of their GPU compatibility known?

**Comment by jabraham17** [comment]
`minloc` and `maxloc` reductions are available on GPU's for arrays

**Comment by e-kayrakli** [comment]
Are they? :) See https://github.com/chapel-lang/chapel/issues/24932. I think minloc/maxloc reductions on GPUs are relatively small amount of effort and we can support them soon enough if that's needed.

In the meantime we have [`gpuMinLocReduce`](https://chapel-lang.org/docs/main/modules/standard/GPU.html#GPU.gpuMinLocReduce) and [`gpuMaxLocReduce`](https://chapel-lang.org/docs/main/modules/standard/GPU.html#GPU.gpuMaxLocReduce) in the GPU module as placeholders.

**Comment by jabraham17** [comment]
I stand corrected. Disappointing 😢

**Comment by bradcray** [comment]
@Iainmon : Since this is covered by #24932, can we close this one?  Or if not, update the title and description to be more about porting the reductions to GPUs?
